lib/upnp/igd_service.go

```diff
@@ -49,12 +50,113 @@ type IGDService struct {
 	ServiceID string
 	URL       string
 	URN       string
-	LocalIP   net.IP
+	LocalIPv4 net.IP
+	Interface *net.Interface
+	PinholeID uint16
+}
```

Comment:
Why doesn't the `IGDService` explicitly include the `nat.Service` interface?  Not really necessary, but a clear statement.

Reply:
I'm sorry, what do you mean by this? I assumed that Go doesn't have a way to explicitly say "this type implements this interface".

Reply:
Simply write `nat.Service` as a member on its own line (preferably with a blank line before).  It means that the struct is checked to actually implement all methods in that interface.

Reply:
Thanks, I learnt something today.